### ISSUE TITLE
Feature: `ifExists` flag for `dropProcedure` 

### ISSUE THREAD

**Issue by nbrugger-tgm** [liquibase/liquibase#5004]
### Search first

- [X] I searched and no similar issues were found

### Description

For most drop operations (like `dropView`) there is a `ifExists` flag to not error if the View or Procedure does not exist. Currently, this has to be done using a `sql` operation, loosing on some of liquibase advantages.

### Steps To Reproduce

Its a "missing feature" so no steps

### Expected/Desired Behavior

When using `ifExists` in a `dropProcedure` operation
1. It does not error on XSD validation
2. When the changeset is run it will not fail if the specified procedure does not exist

### Liquibase Version

any (latest observed: 4.23)

### Database Vendor & Version

-

### Liquibase Extensions

XML changeset

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by kevin-atx** [comment]
@nbrugger-tgm If we provide you with development guidance, would you be interested in submitting a PR to add "if exists" to "drop view"?

**Comment by nbrugger-tgm** [comment]
Yes possibly, altho I can give no time guarantees. Next week or the one after I could spare some time

**Comment by nbrugger-tgm** [comment]
Starting next monday i could take a look since this should be a "relatively simple" change. I will ask for more details if i need it. So you can assign me or whatever marks this as @nbrugger-tgm works on it

**Comment by nbrugger-tgm** [comment]
Thanks for the update - i have not yet done anything since there was not assigned. So this will be done with the linked PR?

**Comment by filipelautert** [comment]
The attached PR will create a default interface to check if a database supports "ifExists something" . Then from there we can implement it. 
I'll go ahead with the mentioned PR and let you know when we can move ahead here. Thanks!

**Comment by filipelautert** [comment]
Hi @nbrugger-tgm ! Did you managed to work on a PR for this? Otherwise I can jump in and implement one.

**Comment by nbrugger-tgm** [comment]
No since you said there is a related pr and I would have to retrofit I waited

**Comment by filipelautert** [comment]
> No since you said there is a related pr and I would have to retrofit I waited

Good! So PR https://github.com/liquibase/liquibase/pull/3581 has been merged - it adds support for `create if not exists`. I think we could create something similar but for `drop if exists`.
